mlrun/datastore/targets.py

```diff
@@ -383,11 +402,21 @@ def update_resource_status(self, status="", producer=None, size=None):
         self._resource.status.update_target(target)
         return target
 
+    def add_writer_step(
+        self, graph, after, features, key_columns=None, timestamp_key=None
+    ):
+        raise NotImplementedError()
```

Comment:
Shouldn't be the same as `add_writer_state` ? why `NotImplementedError` ?

Reply:
That's the implementation from `add_writer_state` - this is a base class (`BaseStoreTarget`) and this method is not implemented here.